dbuf-lsp/src/action_handler/rename.rs

```diff
@@ -22,124 +22,143 @@ pub fn renameable_symbol(symbol: &Symbol) -> bool {
             dependency: _,
         } => true,
         Symbol::Field {
+            t: _,
             constructor: _,
             field: _,
         } => true,
         Symbol::Alias {
             t: _,
             branch_id: _,
-            name: _,
+            alias: _,
+        } => false,
+        Symbol::Constructor {
+            t: _,
+            constructor: _,
         } => false,
-        Symbol::Constructor(_) => false,
         Symbol::None => false,
     }
 }
 
 /// Check if symbol can be renamed to new_name without conflicts.
-pub fn renameable_to_symbol(symbol: &Symbol, new_name: &String, ast: &ElaboratedAst) -> Result<()> {
+pub fn renameable_to_symbol(symbol: &Symbol, new_name: &str, ast: &ElaboratedAst) -> Result<()> {
     if new_name.is_empty() {
-        return rename_errors::rename_to_empty_error();
+        return RenameError::ToEmpty.into();
     }
     if dbuf_language::get_builtin_types().contains(new_name) {
-        return rename_errors::rename_to_builtin_type_error();
+        return RenameError::ToBuiltin.into();
     }
     if dbuf_language::get_keywords().contains(new_name) {
-        return rename_errors::rename_to_keyword_error();
+        return RenameError::ToKeyword.into();
     }
 
     match symbol {
         Symbol::Type(t) => {
-            if dbuf_language::get_builtin_types().contains(t) {
-                return rename_errors::rename_of_builtin_type_error();
-            }
-            if !dbuf_language::is_correct_type_name(new_name) {
-                return rename_errors::rename_to_bad_type_error(new_name);
-            }
             if t == new_name {
-                return rename_errors::rename_to_old_error();
+                return RenameError::ToPrevious.into();
             }
-            if ast.has_type_or_constructor(new_name) {
-                return rename_errors::rename_to_existing_type_error(new_name);
+            if dbuf_language::get_builtin_types().contains(t) {
+                return RenameError::OfBuiltin.into();
             }
-        }
-        Symbol::Dependency {
-            t: type_name,
-            dependency: d,
-        } => {
-            if d == new_name {
-                return rename_errors::rename_to_old_error();
+            let new_type_name: TypeName = match new_name.try_into() {
```

Comment:
You can use `?` here, like this:
```rust
let new_type_name = new_name.try_into().map_err(|_| RenameError::ToExistingType(new_name.to_owned()))?
```

Reply:
Also you could use the default `Result<T, E>`  with `E` set to your own type of errors instead of `jsonrpc::Result` and define all sorts of conversions to your error type from some specific types of errors, like the ones from this `TryFrom` conversion